File: src/main/java/com/employee/system/entity/Leave.java
package com.employee.system.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Leave {
    private Long id;
    private Employee employee;
    private LocalDate startDate;
    private LocalDate endDate;
    private String leaveType; // SICK, CASUAL, EARNED
    private String status; // PENDING, APPROVED, REJECTED, CANCELLED
    private String reason;
    private Integer days;
    private LocalDate createdAt;
    private LocalDate updatedAt;
}

File: frontend/src/types/leave.ts
export interface Leave {
  id: number;
  employee: Employee;
  startDate: string;
  endDate: string;
  leaveType: 'SICK' | 'CASUAL' | 'EARNED';
  status: 'PENDING' | 'APPROVED' | 'REJECTED';
  reason: string;
  days: number;
  createdAt?: string;
  updatedAt?: string;
}

